### PR TITLE
[hf mf sim] Rework - extract to mifaresim.c 

### PR DESCRIPTION
**Pull request by vratiskol**
- extract function from iso14443a.c to mifaresim.c 
- Add **hf mf sim t** on client to select card type (mifare mini, 1k, 2k, 4k)

/!\ Code needs to be reviewed / enhanced to be more stable.

### CODE REVIEW

File: armsrc/mifareutil.c [hunk 1]
```diff
@@ -592,14 +592,27 @@ void emlClearMem(void) {
     memset(emCARD, 0, CARD_MEMORY_SIZE);
 
     // fill sectors trailer data
-    for (uint16_t b = 3; b < 256; ((b < 127) ? (b += 4) : (b += 16)))
+    for (uint16_t b = 3; b <= MIFARE_4K_MAXBLOCK; ((b <= MIFARE_2K_MAXBLOCK) ? (b += 4) : (b += 16)))
```
Comment (doegox) [inline armsrc/mifareutil.c]: I think this goes now out of the buffer when b==MIFARE_4K_MAXBLOCK

Comment (doegox) [reply armsrc/mifareutil.c]: hmm no it's fine as we're jumping over these values and b will never be exactly == MIFARE_4K_MAXBLOCK, still the condition looks a bit weird ;)

**Comment by doegox** [comment]
merged manually to get rid of the client/cmddata.c regression

**Comment by iceman1001** [comment]
Thanks!

**Pull request merged**